tracer/src/Datadog.Trace/Iast/Propagation/DefaultInterpolatedStringHandlerModuleImpl.cs

```diff
@@ -0,0 +1,110 @@
+// <copyright file="DefaultInterpolatedStringHandlerModuleImpl.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#if NET6_0_OR_GREATER
+
+#nullable enable
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+using Datadog.Trace.VendoredMicrosoftCode.System.Runtime.InteropServices;
+
+namespace Datadog.Trace.Iast.Propagation;
+
+internal static class DefaultInterpolatedStringHandlerModuleImpl
+{
+    public static unsafe void Append(IntPtr target, string? value)
```

Comment:
We could consider adding some logic to keep the ranges in some methods such as append without format (like we do with strings). Anyway, this could be done in a separate PR and we should evaluate also the performance cost, so maybe it's not worth it.

Reply:
Do you mean like keeping correct `start` and `length` of tainted data in the ranges?
Currently it's difficult to know the current length before appending the new data to the DefaultInterpolatedStringHandler. The `_pos` int of the struct is private. 😞 
We could get the value by knowing the exact offset in the struct but that would be unsafe (and not done for this PR atm).